### PR TITLE
chore(release): pull release/v1.51.0 into main

### PR DESCRIPTION
**Pull request by devops-github-rudderstack**
:crown: *An automated PR*

### CODE REVIEW

File: src/services/destination/postTransformation.ts [hunk 1]
```diff
@@ -152,6 +161,12 @@ export default class DestinationPostTransformationService {
         authErrorCategory: errObj.authErrorCategory,
       }),
     } as DeliveryResponse;
+
+    // for transformer-proxy to maintain contract
+    const { response } = error;
+    if (response) {
```
Comment (krishna2020) [inline src/services/destination/postTransformation.ts:167]: `DeliveryResponse` + `response` == ErrorResponse 
Do we need two different types ?

File: src/controllers/delivery.ts
Comment (krishna2020) [inline src/controllers/delivery.ts:1]: can we delete `obs.delivery.js` file as  `DestProxyController` also calls `networkHandlerFactory.getNetworkHandler`

File: src/services/delivertTest/deliveryTest.ts [hunk 2]
```diff
@@ -7,7 +7,7 @@ import stats from '../../util/stats';
 import logger from '../../logger';
 import tags from '../../v0/util/tags';
 
-export default class DeliveryTestService {
+export class DeliveryTestService {
   public static async doTestDelivery(
```
Comment (krishna2020) [inline src/services/delivertTest/deliveryTest.ts:11]: how are the test cases passing as `version` should also be passed now ?
https://github.com/rudderlabs/rudder-transformer/pull/2865/files#diff-223989b90529aa3a55fbd07c48aa218b7ece2e1d77fadb70805dbf28f0cd31d5R18

Should add some assertions in proxy destination test cases on whether genericNetworkHandler is expected or not ?

File: src/v0/destinations/marketo/config.js [hunk 3]
```diff
@@ -27,9 +27,9 @@ const formatConfig = (destination) => ({
     false,
   ),
   customActivityPrimaryKeyMap: getHashFromArray(
```
Comment (krishna2020) [inline src/v0/destinations/marketo/config.js:29]: can we make it backward compatible ?
- if `rudderEventsMapping` is empty, continue inferring from `customActivityPrimaryKeyMap`

File: src/services/destination/postTransformation.ts (hunk 1 above)
Comment (utsabc) [reply src/services/destination/postTransformation.ts:167]: cc @aashishmalik

File: src/v0/destinations/marketo/config.js (hunk 3 above)
Comment (utsabc) [reply src/v0/destinations/marketo/config.js:29]: @anantjain45823 can you check this

Comment (anantjain45823) [reply src/v0/destinations/marketo/config.js:29]: Already raised a PR [here](https://github.com/rudderlabs/rudder-transformer/pull/2866)

File: src/controllers/delivery.ts [hunk 4]
```diff
@@ -1,24 +1,30 @@
...
 
-export default class DeliveryController {
+export class DeliveryController {
   public static async deliverToDestination(ctx: Context) {
     logger.debug('Native(Delivery):: Request to transformer::', JSON.stringify(ctx.request.body));
     let deliveryResponse: DeliveryResponse;
     const requestMetadata = MiscService.getRequestMetadata(ctx);
     const event = ctx.request.body as ProcessorTransformationOutput;
     const { destination }: { destination: string } = ctx.params;
+    const { version }: { version: string } = ctx.params;
```
[the remark this answers is not in the document]
Comment (aashishmalik) [reply src/controllers/delivery.ts:19]: will add test cases there, though we use v0 as default version

File: src/services/delivertTest/deliveryTest.ts (hunk 2 above)
Comment (krishna2020) [reply src/services/delivertTest/deliveryTest.ts:11]: Endpoint defaulting to `v0` is fine but the comment is about `getNetworkHandler` call from `deliveryTest.ts`
As version is not being passed, it returns `handlers.generic`

File: src/controllers/delivery.ts
Comment (utsabc) [reply src/controllers/delivery.ts:1]: We have some old tests cases I think for marketo bulk upload which is directly calling it ew can take it up as separate cleanup task as a separate PR

File: src/services/destination/postTransformation.ts (hunk 1 above)
Comment (aashishmalik) [reply src/services/destination/postTransformation.ts]: reverted the typeConversion to `ErrorResponse`, will do this after more tests

File: src/services/destination/postTransformation.ts [hunk 5]
```diff
@@ -12,11 +12,20 @@ import {
...
 } from '../../types/index';
 import { generateErrorObject } from '../../v0/util';
-import ErrorReportingService from '../errorReporting';
+import { ErrorReportingService } from '../errorReporting';
 import tags from '../../v0/util/tags';
 import stats from '../../util/stats';
 
-export default class DestinationPostTransformationService {
+type ErrorResponse = {
+  status?: number;
```
Comment (krishna2020) [inline src/services/destination/postTransformation.ts:20]: can clean up `ErrorResponse`

**Pull request closed without merging**